### ISSUE TITLE
Custom input select for temperature/ventilation mode

### ISSUE THREAD

**Issue by bj00rn** [bj00rn/ha-saleryd-ftx#18]
**Is your feature request related to a problem? Please describe.**
No way to set ventilation/temperature -mode using input select in HA

**Describe the solution you'd like**
Implement custom inout select

**Describe alternatives you've considered**
-

**Additional context**
Inspo: https://github.com/ThermIQ/thermiq_mqtt-ha/blob/91799864a2ee61a6e31711e2c60c34e2007bbc99/custom_components/thermiq_mqtt/input_select.py

**Comment by Fettkeewl** [comment]
Any progress on this? It's a wanted feature 😁

**Comment by bj00rn** [comment]
> Any progress on this? It's a wanted feature 😁

Not sure this is a good idea. Seems that the input_select keeps it's own state in HA somehow which complicates things, and input selects are not really good UX anyways imho. Feel free to dive into the thermiq implementation and do a PR if you wan't.  

At some point i had envisioned some kind of dashboard bundled with the integration, feel free to dive into that too.

You can do a enitity card that calls the set_ventilation_mode service until then:

https://community.home-assistant.io/t/call-service-button-from-lovelace/168656/7

**Comment by bindingofman** [comment]
![Screenshot_20231128_143740_Home Assistant](https://github.com/bj00rn/ha-saleryd-ftx/assets/109304754/5198bf6d-4e5d-4c5a-a8c1-6ab44d118eff)

I use the default settings and all works great for me, this is my card. It's a stock image and is just a representation. I have automations that are connected to temperature and humidity so the ftx will adjust to those conditions. Boost after shower and such.

**Comment by bj00rn** [comment]
> ![Screenshot_20231128_143740_Home Assistant](https://private-user-images.githubusercontent.com/109304754/287196875-5198bf6d-4e5d-4c5a-a8c1-6ab44d118eff.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE0MTc5MjMsIm5iZiI6MTcwMTQxNzYyMywicGF0aCI6Ii8xMDkzMDQ3NTQvMjg3MTk2ODc1LTUxOThiZjZkLTRlNWQtNGM1YS1hOGMxLTZhYjQ0ZDExOGVmZi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwMVQwODAwMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MDA2ZThhYWZjYmY4OWIwYzQ0NmE5Mjk1ODg5NjUzZDQ3ZTM1M2I5NzI0ZWQwYjI2YTdkMzA5MmUwNmM3NDVjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CNHh9eIljxDSRtfTjDgC6sKTz5dHZG49p6vAAer5x_o)
> 
> I use the default settings and all works great for me, this is my card. It's a stock image and is just a representation. I have automations that are connected to temperature and humidity so the ftx will adjust to those conditions. Boost after shower and such.

Sweet! Is it possible to bundle something like this with the integration you think?

**Comment by Fettkeewl** [comment]
> ![Screenshot_20231128_143740_Home Assistant](https://private-user-images.githubusercontent.com/109304754/287196875-5198bf6d-4e5d-4c5a-a8c1-6ab44d118eff.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE1MDY5MDEsIm5iZiI6MTcwMTUwNjYwMSwicGF0aCI6Ii8xMDkzMDQ3NTQvMjg3MTk2ODc1LTUxOThiZjZkLTRlNWQtNGM1YS1hOGMxLTZhYjQ0ZDExOGVmZi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIwMlQwODQzMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWQ4ZTI1YzUyMWQwZTU1NmUyNjAzM2I1YWVkMWVlZmI4N2NlMGIzZWM2MzU1OWUzNzBlZDM2ZjBjYTAyZWUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.P-YS3MtIW1nNQ6mQ0QTQVCCcLrh3jfGxl-LRDRrYAz0)
> 
> I use the default settings and all works great for me, this is my card. It's a stock image and is just a representation. I have automations that are connected to temperature and humidity so the ftx will adjust to those conditions. Boost after shower and such.

Care to share the yaml? I found the image as well. Neat.
Been playing with trying to get animations to work for fans and rotor, no luck sofar :P

**Comment by bindingofman** [comment]
> Care to share the yaml? I found the image as well. Neat. Been playing with trying to get animations to work for fans and rotor, no luck sofar :P

Hello, yes. 
This card is built up of custom:stack-in-card, custom:mushroom, and custom:button-card (this is the one for animating icons)
```yaml
  - type: horizontal-stack
    cards:
      - type: custom:button-card
        entity: switch.saleryd_hrv_away_mode
        name: Away
      - type: custom:button-card
        name: Boost
        entity: switch.saleryd_hrv_boost_mode
        state:
          - value: 'on'
            color: lightblue
            styles:
              icon:
                - animation: rotating 2s linear infinite
          - value: 'off'
            styles:
              icon:
                - animation: rotating 6s linear infinite
                 
```

The boost icon is revolving even when off, but when activated it changes color and goes faster.


```yaml
  - type: horizontal-stack
    cards:
      - type: custom:button-card
        tap_action:
          action: call-service
          service: saleryd_hrv.set_temperature_mode
          target: {}
          data:
            value: 0
        icon: mdi:thermometer-chevron-up
        entity: sensor.saleryd_hrv_target_temperature
        state:
          - value: '21'
            color: light blue
          - value: '19'
            color: rgba(68,115,158)
          - value: '16'
            color: rgba(68,115,158)
        name: 21°C

```

Set temp icons calls the service for the 3 available temperatures.
This could/ probably should be tidied up, it was a year since I got the Saleryd machine and added the integration.

**Comment by Fettkeewl** [comment]
Thanks for sharing.
Getting there slowly, will play with colors later :)
![image](https://github.com/bj00rn/ha-saleryd-ftx/assets/25577040/59c15c27-07b8-41e3-af2a-85747dbe51fc)

**Comment by bj00rn** [comment]
Still figuring out how to bundle frontend controls with integration.

Made a minimal example in https://github.com/bj00rn/ha-saleryd-ftx/pull/31

Have a look at the alias section to define entities for extra sensors

**Comment by bj00rn** [comment]
> Well we are on the right path here :P this is my initial draft in a ow so tired state after work and putting kids to bed... Will try to smooth things out later ![Screenshot_20231201_203741_Home Assistant](https://private-user-images.githubusercontent.com/25577040/287365019-e44d9576-41be-457f-a52f-0f161a4fc78e.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDM0MTU1OTAsIm5iZiI6MTcwMzQxNTI5MCwicGF0aCI6Ii8yNTU3NzA0MC8yODczNjUwMTktZTQ0ZDk1NzYtNDFiZS00NTdmLWE1MmYtMGYxNjFhNGZjNzhlLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjI0VDEwNTQ1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU0YjRmYjU5Y2M4ZjM0NmNiNzhkMDVjNTExMzVhODBhY2Y2M2JmMDM3ZGViYzQyMDIwOTY5NzVmZjNkNGRlMDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HEFCMdrLyn9RJ5wz1zWKyLIROm3e2pJgXyUAKughayY)

Do you have a power meter connected to the FTX? Could you try latest release 3, and confirm the that the heater power calculation is correct? Seems a bit low to me. Fans seem to be rated at around 182W at 100%, heater is 900W at 100% and the system is rated at 1100W nominal.

**Comment by Fettkeewl** [comment]
@bj00rn hope this helps
Also it was calibrated using a dumb meter
![i1](https://github.com/bj00rn/ha-saleryd-ftx/assets/25577040/82447eb2-5a17-4250-b3b8-121439413efc)
![i2](https://github.com/bj00rn/ha-saleryd-ftx/assets/25577040/40594cb2-7cf7-4d02-a5da-cfe56b26cba9)
![i3](https://github.com/bj00rn/ha-saleryd-ftx/assets/25577040/c7cfdbac-17a4-4dd6-a062-d78fb7bd7c92)

**Comment by Fettkeewl** [comment]
Yea no sorry @bj00rn  the % does not translate to an active power. 
Please redo the change : )
The percentage must mean how long the heater is "ON" for a cycle if I am to guess. 
The heater turns full throttle i'd guess'timate and is on for 29% of an cycle which appears to be something in the neighbourhood of 4½-5h

![image](https://github.com/bj00rn/ha-saleryd-ftx/assets/25577040/70be61f9-3317-4f28-bb1d-6d4dbe877f3e)

**Comment by bj00rn** [comment]
@Fettkeewl thanks! A bit more complicated than I imagined then 😪.

**Comment by bj00rn** [comment]
@Fettkeewl after some thought i still feel like the power in W is relevant here, since it would correspond to an estinated average heater power over a cycle of any given length.

Example: If 100% equates to 800W constant power. 50% would be 800W at 50% of a cycle = an avarage of 400W over the cycle.

**Comment by Fettkeewl** [comment]
@bj00rn Can you atleast readd the previous sensor? =) I liked it better! And its visible on the Saleryds own webpage 
so I for one would want to have it around 🗡️

**Comment by bj00rn** [comment]
> @bj00rn Can you atleast readd the previous sensor? =) I liked it better! And its visible on the Saleryds own webpage so I for one would want to have it around 🗡️

Which one do you mean I should readd? The one that was constant 800W? I renamed heater_temperature_percent -> heater_ power_percent. And added new sensor heater_power, that replaced the constant 800W.

**Comment by Fettkeewl** [comment]
My bad missed it, wanted the percentage sensor back 🙏

**Comment by bj00rn** [comment]
fixed by #57
